### ISSUE TITLE
Pi.Alert taking a long time to start with UID/GID defined

### ISSUE THREAD

**Issue by acbgbca** [jokob-sk/Pi.Alert#130]
**Describe the issue**

Pi.Alert is taking a long time to start for me, sometimes several minutes. After some investigation I believe the problem is related to the 'chown' command in user-mapping.sh (line 27)

Investigating further, the reason that command takes so long is that there are '9261' files in the /home/pi/pialert directory:

```
# ls -1 -R . | wc -l
9261
```

The vast majority of files appear to be under /home/pi/pialert/front/lib/AdminLTE (9048), and bower_components within that (8499)

I think the main issue is that the chown should not be required. Unless those files are being edited and/or being stored in a volume, the owner isn't important provided they can be read. Given they are linked under /var/www/html, the owner likely should be www-data.

I believe the simplest solution would be to have the Dockerfile put those files directly under /var/www/html instead of using a symbolic link, then they would not have the owner changed.

**Paste last few lines from `pialert.log`**
n/a

**Paste your `pialert.conf` (remove personal info)**
n/a

**Paste your `docker-compose.yml` and `.env` (remove personal info)**

`docker-compose.yml` 
```

---
version: "2.4"
services:
  pialert:
    container_name: pialert
    image: "jokobsk/pi.alert:latest"      
    network_mode: "host"        
    restart: unless-stopped
    volumes:
      - ./config:/home/pi/pialert/config
      - ./db/pialert.db:/home/pi/pialert/db/pialert.db
      - ./db/setting_darkmode:/home/pi/pialert/db/setting_darkmode
      - ./log:/home/pi/pialert/log
    environment:
      - TZ=Australia/Melbourne
      - PORT=20211
      - HOST_USER_ID=9001
      - HOST_USER_GID=9001

```

`.env` 
none

**Screenshots**

**Comment by acbgbca** [comment]
I have some experience with creating docker images, if you like my solution I would be happy to make that change, and maybe see if there is anything else that can be cleaned up.

**Comment by jokob-sk** [comment]
Hi There,

Yes, ownership of files is all over the place. 

Feel free to submit a PR limiting the files affected by the chown by limiting it's scope to the directories which are mountable, so probably /db /config , not sure about the /front one. I do mount /front for development purposes so not sure if that matters. 

I'd definitely don't go down the path changing the structure too much as a lot of code depends on files being in the right place. 

E.g.:

https://github.com/jokob-sk/Pi.Alert/blob/1ea7990314f3be1b42016e54f67e2d47fe972d71/front/maintenance.php#L122

I know this is not ideal, but a lot of it is legacy code and not sure what's the value in refactoring it. So changing and moving files around would probably than require rewrites/more testing. 

So in short, definitely any PR improving the image is welcome, however any change should be backwards compatible and shouldn't require too much code rewrites.  Not sure if speeding up the first run is worth this much effort. Also not 100% sure if this is necessarily slowing down the boot up the most/significantly. Just something to be aware of I guess.

Thanks in advance!
j

**Comment by acbgbca** [comment]
I see what you mean, the permissions are a bit of a mess.

I was able to quantify the start time impacts of the chown; 7m52s on my machine (Intel i5, about 5 years old)

When I get some time I'll look into this further. If the users and groups are set correctly it should be possible to only change permissions on the mounted volumes, and leave everything else alone.

**Comment by devlincashman** [comment]
I think you should be able to remove most of the user mapping script functionality, as the chown and chmod COPY in the recent dockerfile PR will allow nginx to read/write all of the symlinked html and php. This is a bit of a kluge, but as long as all of the relevant user id's (ie root, www-data and pi) are all part of the pi group they should have full r/w/x.

**Comment by jokob-sk** [comment]
Will reopen if some work happens in this regard
